nengo_loihi/simulator.py

```diff
@@ -602,20 +602,11 @@ def handle_chip2host_communications(self):  # noqa: C901
                 for cx_probe, probe in self.cx_probe2probe.items():
                     x = data[snip_range[cx_probe]]
                     if cx_probe.key == 's':
-                        if isinstance(probe.target, nengo.ensemble.Neurons):
-                            n_type = probe.target.ensemble.neuron_type
-                            if hasattr(n_type, 'tau_ref'):
-                                t_ref = n_type.tau_ref
-                                ref_period_in_steps = int(t_ref/self.dt)
-                            else:
-                                ref_period_in_steps = 0
-
-                            # loihi uses the voltage value to indicate
-                            #  where we are in the refractory period
-                            x = (x == (ref_period_in_steps + 1) * 128)
-                        else:
-                            # interneurons have a refractory period of 1 step
-                            x = (x == 128)
+                        # Loihi uses the voltage value to indicate where we
+                        # are in the refractory period. We want to find neurons
+                        # starting their refractory period.
+                        refract_delays = cx_probe.target.refractDelay
+                        x = (x == refract_delays * 128)
```

Comment:
Nice!  So much cleaner.  Thank you!

Reply:
I wasn't quite done. I've added another commit for the case that `cx_probe.target` is not a CxGroup. I'm not sure if this ever actually happens, but it currently is possible to have a CxProbe on a CxSpikeInput (though I think we handle that all on the CPU side, which is probably why it never happens).